### PR TITLE
Add namespace scope for Ray cluster get and delete operations 

### PR DESCRIPTION
**Pull request by daikeshi**
## Why are these changes needed?
This PR added namespaced operations to get and delete a Ray cluster via KubeRay API server and CLI

## Related issue number
Closes https://github.com/ray-project/kuberay/issues/209

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: apiserver/pkg/manager/resource_manager.go [hunk 1]
```diff
@@ -20,9 +20,9 @@ const DefaultNamespace = "ray-system"
 // kubernetes objects and potential db objects underneath operations should be encapsulated at this layer
 type ResourceManagerInterface interface {
 	CreateCluster(ctx context.Context, apiCluster *api.Cluster) (*v1alpha1.RayCluster, error)
-	GetCluster(ctx context.Context, clusterName string) (*v1alpha1.RayCluster, error)
+	GetCluster(ctx context.Context, clusterName string, namespace string) (*v1alpha1.RayCluster, error)
 	ListClusters(ctx context.Context) ([]*v1alpha1.RayCluster, error)
```
Comment (Jeffwan) [inline apiserver/pkg/manager/resource_manager.go:24]: what's the plan for the list operation? I assume user may not have access to list cluster across namespaces? Should we limit to namespace scope?

File: apiserver/pkg/manager/resource_manager.go [hunk 2]
```diff
@@ -104,12 +104,12 @@ func (r *ResourceManager) populateComputeTemplate(ctx context.Context, cluster *
...
 }
 
-func (r *ResourceManager) GetCluster(ctx context.Context, clusterName string) (*v1alpha1.RayCluster, error) {
+func (r *ResourceManager) GetCluster(ctx context.Context, clusterName string, namespace string) (*v1alpha1.RayCluster, error) {
 	if len(clusterName) == 0 {
 		return nil, util.NewInvalidInputError("clusterName is empty, failed to get the cluster.")
 	}
 
-	client := r.getRayClusterClient(DefaultNamespace)
+	client := r.getRayClusterClient(namespace)
```
Comment (Jeffwan) [inline apiserver/pkg/manager/resource_manager.go:112]: 1. Do we want to support default namespace? 
2. Is namespace a required field?

Comment (Jeffwan) [reply apiserver/pkg/manager/resource_manager.go:112]: I was thinking whether we like to support backward compatibility? In our interval version, `namespace` is still optional, if user doesn't specify it, we store the CR in the default namespace. If they do, we then use the namespace. In upstream version, this is still under active development, so I think either way works.

File: apiserver/pkg/manager/resource_manager.go (hunk 1 above)
Comment (daikeshi) [reply apiserver/pkg/manager/resource_manager.go:24]: Yeah I agree. Having namespace scope for the list op makes more sense here. 

I can work on this after this PR is merged.

File: apiserver/pkg/manager/resource_manager.go (hunk 2 above)
Comment (daikeshi) [reply apiserver/pkg/manager/resource_manager.go:112]: >Is namespace a required field?

Namespace is a required field in protos, but api server set the default namespace when it's empty for `create` op. I'll follow the existing logic for `get` and `delete` ops as well.

>Do we want to support default namespace?

I think either way would work for us. Maybe we can make this default namespace configurable in the future.

Comment (daikeshi) [reply apiserver/pkg/manager/resource_manager.go:116]: Set `namespace` to `DefaultNamespace` [here](https://github.com/ray-project/kuberay/pull/229/files#diff-6d24c4f57fa68eaf6e5d018e13cb8549476d9047c9bd3cf8f496a5760e8bc3b9R112)

File: apiserver/pkg/manager/resource_manager.go (hunk 1 above)
Comment (Jeffwan) [reply apiserver/pkg/manager/resource_manager.go:24]: sounds good. Please help create an issue to track it later.

**Review by Jeffwan — APPROVED** [review]
The changes look good to me. Thanks for the contribution!

**Pull request merged**